modules/javafx.graphics/src/main/java/com/sun/glass/ui/headless/HeadlessWindow.java

```diff
@@ -0,0 +1,370 @@
+        this.alpha = alpha;
+    }
+
+    @Override
+    protected boolean _setBackground(long ptr, float r, float g, float b) {
+        this.bg_r = r;
+        this.bg_g = g;
+        this.bg_b = b;
+        return true;
+    }
+
+    @Override
+    protected void _setEnabled(long ptr, boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    @Override
+    protected boolean _setMinimumSize(long ptr, int width, int height) {
+        this.minWidth = width;
+        this.minHeight = height;
+        return true;
+    }
+
+    @Override
+    protected boolean _setMaximumSize(long ptr, int width, int height) {
+        this.maxWidth = width;
+        this.maxHeight = height;
+        return true;
+    }
+
+    @Override
+    protected void _setIcon(long ptr, Pixels pixels) {
+        this.icon = pixels;
+    }
+
+    @Override
+    protected void _setCursor(long ptr, Cursor cursor) {
+        this.cursor = cursor;
+    }
+
+    @Override
+    protected void _toFront(long ptr) {
+    }
+
+    @Override
+    protected void _toBack(long ptr) {
+    }
+
+    @Override
+    protected void _enterModal(long ptr) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    protected void _enterModalWithWindow(long dialog, long window) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    protected void _exitModal(long ptr) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    protected void _requestInput(long ptr, String text, int type, double width, double height, double Mxx, double Mxy, double Mxz, double Mxt, double Myx, double Myy, double Myz, double Myt, double Mzx, double Mzy, double Mzz, double Mzt) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    protected void _releaseInput(long ptr) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    boolean setFullscreen(boolean full) {
+        int newX = 0;
+        int newY = 0;
+        int newWidth = 0;
+        int newHeight = 0;
+        if (full) {
+            this.originalHeight = this.height;
+            this.originalWidth = this.width;
+            this.originalX = this.x;
+            this.originalY = this.y;
+            newX = 0;
+            newY = 0;
+            newWidth = screen.getWidth();
+            newHeight = screen.getHeight();
+        } else {
+            newHeight = this.originalHeight;
+            newWidth = this.originalWidth;
+            newX = this.originalX;
+            newY = this.originalY;
+        }
+        notifyResizeAndMove(newX, newY, newWidth, newHeight);
+        return full;
+    }
+
+    private void notifyResizeAndMove(int x, int y, int width, int height) {
+        HeadlessView view = (HeadlessView) getView();
+        //   if (getWidth() != width || getHeight() != height) {
```

Comment:
Why is this code commented out?

Reply:
There was a failing test, so it turns out there is an implicit contract that even if a resize is requested without different w/h parameters, at least the even should still be generated.

Reply:
should we either add a comment or clean up (or both) then?

Reply:
I cleaned it up. Typically, I'm +1 on adding more comments in the code, but not everyone likes that. In this case, it would create a precedent where we would then add tons of comments mainly to document what the tests are doing (or expecting).